cpp/src/io/parquet/page_string_decode.cu

```diff
@@ -663,38 +663,19 @@ __global__ void __launch_bounds__(decode_block_size) gpuDecodeStringPageData(
   page_state_buffers_s* const sb = &state_buffers;
   int const page_idx             = blockIdx.x;
   int const t                    = threadIdx.x;
+  [[maybe_unused]] null_count_back_copier _{s, t};
```

Comment:
How does this avoid the race condition when two separate kernels visit the same page? Won't one of them erroneously zero the page out that another may have written a valid value to?

Reply:
Only one invocation should make it past the filter. That one will zero out the null count and then the back copier will copy it back to the page. @vuule added the logic to make the back copy a no-op if the setup returns early.

Reply:
Ah, I see.  Checking to see if the nesting_info pointer is null.